src/librustdoc/html/static/main.js

```diff
@@ -37,10 +37,58 @@
                      "associatedconstant",
                      "union"];
 
+    function hasClass(elem, className) {
+        if (elem && className && elem.className) {
+            var x = elem.className.split(' ');
+            for (var i = 0; i < x.length; i++) {
+                if (x[i] === className) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    function addClass(elem, className) {
+        if (elem && className && !hasClass(elem, className)) {
+            if (elem.className && elem.className.length > 0) {
+                elem.className += ' ' + className;
+            } else {
+                elem.className = className;
+            }
+        }
+    }
+
+    function removeClass(elem, className) {
+        if (elem && className && elem.className) {
+            var x = elem.className.split(' ');
+            for (var i = 0; i < x.length; i++) {
+                if (x[i] === className) {
+                    x.splice(i, 1);
+                }
+            }
+            elem.className = x.join(' ');
+        }
+    }
+
+    function on_each(arr, func) {
```

Comment:
It's not available because many of the methods used, such as getElementsByClassName(), don't return an array, but an array-like object such as [HTMLCollection](https://developer.mozilla.org/en-US/docs/Web/API/HTMLCollection), which indeed doesn't implement forEach(). :/